src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaUserService.java

```diff
@@ -130,6 +134,46 @@ public void updateLastAccessed(User user) {
 	}
 
 	@Override
+	public List<User> findAllByWorkgroupAndRoleToken(Workgroup workgroup, String roleToken) {
+		List<User> users = new ArrayList<>();
+
+		List<UserRole> userRoles = workgroup.getUserRoles();
+		for (UserRole userRole : userRoles) {
+			if (userRole.getRoleToken().equals(roleToken)) {
+				users.add(userRole.getUser());
+			}
+		}
+
+		return users;
+	}
+
+	@Override
+	public List<User> findAllByLineItems(List<LineItem> lineItems) {
+		List<User> users = new ArrayList<>();
+		List<String> addedLoginIds = new ArrayList<>();
+
+		for (LineItem lineItem : lineItems) {
+			String modifiedBy = lineItem.getModifiedBy();
+			// modifiedBy will come in the form "system" or "user:guilden"
+			if (modifiedBy.indexOf(":") > 0) {
```

Comment:
This seems like LineItem logic. Maybe a transient helper function to get the loginId instead of this potentially reusable logic being in UserService?